reports/report-models.js

```diff
@@ -1,26 +1,31 @@
 var mongoose = require('mongoose');
 var Schema = mongoose.Schema;
 
-var eventReportSchema = new Schema({
-  type: String,
-  approvalStatus: Boolean,
-  student: String,
-  event: String,
-})
-
-var otherReportSchema = new Schema({
-  type: String,
-  approvalStatus: Boolean,
-  student: String,
-  category: String,
-  datetime: Date,
-  location: String,
-  title: String,
-  description: String
-})
-
-var EventReport = mongoose.model('EventReport', eventReportSchema, 'reports');
-
-var OtherReport = mongoose.model('OtherReport', otherReportSchema, 'reports');
-
-module.exports = { EventReport, OtherReport };
+var options = {discriminatorKey: 'type'};
+
+var reportSchema = new Schema(
+    {
+      approvalStatus: Boolean,
+      student: String,
+    },
+    options);
+
+var eventReportSchema = new Schema({event: String}, options);
+
+var otherReportSchema = new Schema(
+    {
+      category: String,
+      datetime: Date,
+      location: String,
+      title: String,
+      description: String,
+    },
+    options);
+
+var Report = mongoose.model('Report', reportSchema);
+
+var EventReport = Report.discriminator('EventReport', eventReportSchema);
+
+var OtherReport = Report.discriminator('OtherReport', otherReportSchema);
+
+module.exports = {Report, EventReport, OtherReport};
```

Comment:
Report is now being used for the RUD portion of CRUD, since it provides a generic way to access both EventReports and OtherReports already in the database.  Before using discriminators, I was just using EventReport to find() reports, but if you try to update OtherReport fields from that, it doesn't work, because EventReport only returns objects that you can treat like EventReports (it's all really weird).